torch/fx/experimental/proxy_tensor.py

```diff
@@ -629,12 +630,7 @@ def wrap_fake_symbolic(x, sym_shape):
 
 
 def get_torch_dispatch_modes():
-    modes = [torch._C._get_torch_dispatch_mode()]
-    if modes[-1] is None:
-        return list()
-    while modes[-1].inner is not None:
-        modes.append(modes[-1].inner)
-    return modes
+    return torch.utils._python_dispatch._cur_torch_dispatch_mode
```

Comment:
TODO(samdow): return a copy, not the actual list since updating the list in place will update the mode stack itself. Add a function to return all the modes that returns a copy. Document that it's references to the modes but a copy of the stack

Reply:
Added comments here and in the _python_dispatch and overrides files